### PR TITLE
Add/clarify OAI-PMH documentation

### PR DESCRIPTION
**Pull request by kspurgin**
## Purpose / why

Reading the OAI-PMH documentation page yesterday did not answer a couple of questions I had. I asked those questions in Slack. After seeing the answers in Slack, I realized part of the answers were implicit in the existing documentation page, but not really clear if you are coming to this fresh.

## What changes were made?

- Adds two new sections to the OAI-PMH doc page: `OAI-DC Metadata Format/Metadata Mappings` and `Creating additional metadata formats`
- Reorganizes some of the existing documentation content into these two sections, and adds some additional information to them
- Clarifies that the "OAI Dublin Core (RDF Mapping)" you see in the OAI config is not a separate RDF mapping 
- Adds link to example metadata format plugin

## Verification??

These edits are based on responses from @seth-shaw-unlv and @jordandukart in Slack ([here](https://islandora.slack.com/archives/CM5PPAV28/p1646767998263429) and [here(thread)](https://islandora.slack.com/archives/CM5PPAV28/p1646768064514379)

They may wish to verify I have not misrepresented their responses in working them into the documentation page. 

## Interested Parties

@Islandora/documentation

## Checklist

> __Pull-request reviewer__ should ensure the following

* [ ] Does this PR link to related [issues](https://github.com/Islandora/documentation/issues/)?
* [ ] Does the proposed documentation align with the [Islandora Documentation Style Guide](https://islandora.github.io/documentation/contributing/docs_style_guide/)?
* [ ] Are the changes accurate, useful, free of typos, etc?

> __Person merging__ should ensure the following
* [ ] Does mkdocs still build successfully? (This is indicated by TravisCI passing. To test locally, and see warnings, see [How To Build Documentation](https://islandora.github.io/documentation/technical-documentation/docs-build/).)
* [ ] If pages are renamed or removed, have all internal links to those pages been fixed?
* [ ] If pages are added, have they been linked to or placed in the menu?
* [ ] Did the PR receive at least one approval from a committer, and all issues raised have been addressed?

### CODE REVIEW

**Comment by seth-shaw-unlv** [comment]
The note about creating a profile via the twig templating system predates the metadata map plugins; I would scratch that. I would also move the note about the metatag plugin into the "OAI-DC Metadata Format/Metadata Mappings" section as the RDF and Metatag plugins are both ways of getting defining OAI-DC.

File: docs/user-documentation/oai.md [hunk 1]
```diff
@@ -17,6 +17,23 @@ Enabling Islandora OAI-PMH provides:
...
+
+Note that this is not a separate RDF mapping. It uses the RDF mapping for the included content type(s) mapping (e.g. rdf.mapping.node.islandora_object.yml) and filters out any fields where the predicate is not in the Dublin Core namespace.
+
+!!! tip "Field values not showing up in OAI-DC record?"
+    If you want the value of a field to be emitted in the OAI-DC record, you must assign a Dublin Core predicate for that field in your content type's RDF mapping. If you are wondering why a field is not showing up in the OAI-DC record, the content type's RDF mapping is the first thing to check. 
+
+The OAI-DC metadata format is aided by the Islandora OAI-PMH module to include the RDF Mapping for the linked agent field (from Islandora Defaults). Including agent links in the OAI-PMH metadata currently requires updating the RDF mapping to include a Dublin Core predicate for that field or any other additional fields.
+
+## Creating additional metadata formats
+This typically involves creating a new plugin.
```
Comment (jordandukart) [inline docs/user-documentation/oai.md]: Really nitpicking here but think for accuracy AFAIK it _does_ require creating a new plugin to support a new metadata format.

File: docs/user-documentation/oai.md [hunk 2]
```diff
@@ -17,6 +17,23 @@ Enabling Islandora OAI-PMH provides:
...
+
+The [Drupal rest_oai_pmh module's DefaultMap plugin](https://git.drupalcode.org/project/rest_oai_pmh/-/blob/2.0.x/src/Plugin/OaiMetadataMap/DefaultMap.php) provides a basic model to follow for creating a plugin. 
+
+Exact implementation of your plugin will depend on your data model. If your data model uses anything like Typed Relation field types, Paragraphs, or other complex nested entity modeling, you will need to add custom logic to build the values to emit via OAI-PMH for those fields.
```
Comment (jordandukart) [inline docs/user-documentation/oai.md]: Not sure the best way to present / convey this for documentation but a lot of the default `rest_oai_pmh` conventions expect a very flat render array style of fields to be returned when implementing [`transformRecord()`](https://git.drupalcode.org/project/rest_oai_pmh/-/blob/2.0.x/src/Plugin/OaiMetadataMap/DublinCoreRdf.php#L116) unless you are really going to go off the beaten path with introducing a custom twig template or build method.

Guess tl;dring the above would be the default processes expect a flat array of fields to be returned?

File: docs/user-documentation/oai.md (hunk 1 above)
Comment (kspurgin) [reply docs/user-documentation/oai.md]: That's not nitpicking. I was confusing defining mappings for OAI-DC with defining a mapping in general. Fixed.

File: docs/user-documentation/oai.md (hunk 2 above)
Comment (kspurgin) [reply docs/user-documentation/oai.md]: Yeah, I think anyone with any experience writing any code for Islandora will get the drift. I feel like I even ran into this basic idea writing field display stuff for I7.  Tweaked the wording a bit.

**Comment by kspurgin** [comment]
Feedback incorporated.

**Comment by ysuarez** [comment]
@kspurgin, speaking on behalf of DIG, are you planning on making changes do this PR, are you waiting for someone to review, or do you need any help from DIG to get this merged into the documentation? Just trying to help. Thanks in advance.

**Comment by kspurgin** [comment]
Thanks for checking in, @ysuarez. 

I don't plan on making any more changes unless further changes are requested. I believe I incorporated all existing feedback from @seth-shaw-unlv and @jordandukart

**Review by seth-shaw-unlv — APPROVED** [review]
👍

**Pull request merged**